models/netsuite2/stg_netsuite2__transactions.sql

```diff
@@ -27,17 +27,20 @@ final as (
         transactionnumber as transaction_number,
         type as transaction_type,
         memo,
-        trandate as transaction_date,
+        cast(trandate as date) as transaction_date,
         status,
         createddate as created_at,
-        duedate as due_date_at,
+        cast(duedate as date) as due_date_at,
         closedate as closed_at,
```

Comment:
When spot checking these transaction fields which weren't changed in this PR in the [Netsuite documentation](https://www.netsuite.com/help/helpcenter/en_US/srbrowser/Browser2021_1/analytics/record/transaction.html) I can see that it looks like `createddate` is listed as a datetime and shouldn't be adjusted. However, `closedate` does look to be a date. Is this how it reflects in the Fivetran synced data? Can we take a closer look at which fields should and shouldn't be converted.

![image](https://github.com/user-attachments/assets/c55f0648-cbb9-486f-9191-550b1e9bf9ba)

Reply:
Good callout. [Based on Jared's PR](https://github.com/fivetran/dbt_netsuite_source/pull/54), it looks as if he was only concerned with columns in the Netsuite UI that weren't performing timezone conversion rather than providing explicit casting of all columns to dates that were originally timestamps. 

I'm not sure if this is outside the scope of this issue though. Should I create another issue to address this or batch this into the exiting one and bring everything in?

(I do wonder if it's best to point out these updates with finance as well, just to see if there are any concerns with casting these fields to dates).

Reply:
(`closedate` is itself a timestamp in the Fivetran data.)

Reply:
Below is the reasoning from the PR on casting specific fields as `date`. 

> Keeping these fields as timestamp causes issues in reporting tools that perform automatic timezone conversion.

Example: 
Someone in GMT +2 would see 8/1 when the NetSuite UI shows 7/31 for a value of `2024-07-31 00:00:00.000000000 +00:00`. The is particularly a problem for fields like `ending_at`.

I only CAST fields that were causing problems. However, you are safe to include others like `closedate` where the documentation states `date`. We would not want to include `createddate` or other fields the documentation lists as `datetime`. The NetSuite UI presents timestamp info for those fields, and perform timezone conversion.

Reply:
> (closedate is itself a timestamp in the Fivetran data.)

It's a timestamp, however, the time is always midnight UTC. This indicates is should be treated as a `date`. NetSuite docs also list it as a `date`.

Reply:
Thanks for the context @jmongerlyra ! 

So is it safe to assume that if we see a field that is in timestamp format, but the timestamp is always in 00:00:00 format for that field, it's safe to treat it as a `date` and cast it as such?

Reply:
@fivetran-avinash Midnight UTC values combined with ensuring the NetSuite documentation has the field listed as `date` means it should be safe to CAST as `date`.

Reply:
Thanks @jmongerlyra . Based on that advice and reading of the documentation, I've cast the following additional columns as `date`:

- `firstorderdate` with `customers` (`date_first_order_at` in `stg_netsuite2__customers`)
- `closedate` with `transactions` (`closed_at` in `stg_netsuite2__transactions`)

Every other timestamp had non-00:00:00 format in our development data, or documentation indicated it was datetime. 

(Also `hire_date` would have been a candidate in `employees`, but it was not brought into the staging model)